lib/publiccloud/provider.pm

```diff
@@ -230,19 +254,17 @@ sub create_instance {
     my ($self, %args) = @_;
     $args{check_connectivity} //= 1;
 
-    my @vms      = $self->terraform_apply(%args);
-    my $instance = $vms[0];
-    record_info('INSTANCE', Dumper($instance));
-
-    if ($args{check_connectivity}) {
-        $instance->check_ssh_port();
+    my @vms = $self->terraform_apply(%args);
+    foreach my $instance (@vms) {
+        record_info("INSTANCE $instance->{instance_id}", Dumper($instance));
+        $instance->check_ssh_port() if ($args{check_connectivity});
     }
-    return $instance;
+    return @vms;
```

Comment:
Hmm I have to check this and yes create a new function for this.

Reply:
You could also use [wantarray](https://perldoc.perl.org/functions/wantarray.html) and have it both ways:

```
return wantarray ? @vms : $instance;
```

Reply:
@alvarocarvajald Thx a lot!

Reply:
There are folks in openqa who don't like wantarray, as it is considered as bad practice. 
e.g. https://github.com/os-autoinst/os-autoinst/pull/1165#pullrequestreview-246620436 ff.
Thats why I proposed new function.

Reply:
If it's considered bad practice within the project, better to avoid it then. Here's a nice discussion from 11 years ago regarding pros and cons: https://www.perlmonks.org/?node_id=729965 though.

I suppose your proposed workaround from https://github.com/os-autoinst/os-autoinst/pull/1165#pullrequestreview-246620436, i.e., `$args{wantarray}` could be a good compromise instead of writing a separate method that would be mostly the same. WDYT?

Reply:
Oh, that I like.